web/yo/app/scripts/directives/reviewpanel.js

```diff
@@ -171,15 +179,37 @@ angular.module('oncokbApp')
                             items = [{reviewObj: $scope.obj.type_review, uuid: $scope.obj.type_uuid}];
                             break;
                         case 'ONCOGENIC':
-                            uuid = $scope.mt.oncogenic_uuid;
+                            var oncogenicUUID = $scope.mt.oncogenic_uuid.getText();
+                            var effectUUID = $scope.mt.effect_uuid.getText();
+                            $scope.oncogenicLastUpdateTime = new Date().toLocaleString();
+                            $scope.effectLastUpdateTime = new Date().toLocaleString();
                             items = [{obj: $scope.mt.oncogenic, reviewObj: $scope.mt.oncogenic_review, uuid: $scope.mt.oncogenic_uuid},
-                                {obj: $scope.mt.summary, reviewObj: $scope.mt.summary_review, uuid: $scope.mt.summary_uuid},
-                                {obj: $scope.mt.shortSummary, reviewObj: $scope.mt.shortSummary_review, uuid: $scope.mt.shortSummary_uuid}];
-                            break;
-                        case 'MUTATION_EFFECT':
-                            uuid = $scope.mt.effect_uuid;
-                            items = [{obj: $scope.mt.effect.value, reviewObj: $scope.mt.effect_review, uuid: $scope.mt.effect_uuid},
+                                {obj: $scope.mt.effect.value, reviewObj: $scope.mt.effect_review, uuid: $scope.mt.effect_uuid},
                                 {obj: $scope.mt.description, reviewObj: $scope.mt.description_review, uuid: $scope.mt.description_uuid}];
+
+                            var oncogenicChange = mainUtils.needReview($scope.mt.oncogenic_uuid);
+                            var effectChange = mainUtils.needReview($scope.mt.effect_uuid) || mainUtils.needReview($scope.mt.description_uuid);
+                            if(oncogenicChange && effectChange) {
+                                $scope.bothChanged = true;
+                                DatabaseConnector.getEvidencesByUUIDs([oncogenicUUID, effectUUID], function(result) {
+                                    var resultJSON = JSON.parse(result.status);
+                                    if (_.isArray(resultJSON)) {
+                                        if(resultJSON[0]) {
+                                            setUpdateTimeEffectSection(resultJSON[0], oncogenicUUID, effectUUID);
+                                        }
+                                        if(resultJSON[1]) {
+                                            setUpdateTimeEffectSection(resultJSON[1], oncogenicUUID, effectUUID);
+                                        }
+                                    }
+                                    specialCases();
```

Comment:
Why do we go through `specialCases` for ONCOGENIC? I didn't see any switch case inside of that function